src/OSInfo.cpp

```diff
@@ -23,6 +23,20 @@
 #include "Version.h"
 
 #if defined(Q_OS_WIN)
+// regString converts a wchar_t string of size to a
+// QString. If the string contains a NUL value, that
+// NUL value will terminate the string.
+static QString regString(wchar_t *string, int size) {
+	if (size <= 0) {
```

Comment:
If they don't handle == 0 then I would consider it a doc bug. Just reading those they ought to handle it just fine. <0 is another thing altogether as wcsnlen takes a size_t so no meaningful thing would happen with that ;) But checking both is 100% fine, I just wanted to make sure I didn't misread the docs so I don't produce potential bugs by misusing them under wrong assumptions going forwards.